accepted/0000-salt-new-maintaining-workflow.md

```diff
@@ -0,0 +1,252 @@
+   * services enabled
+   * package building based on `openSUSE/release/xxxx` GitHub branch.
+- `systemsmanagement:saltstack:github:CI:...:PR-XXXX/salt`:
+   * services enabled
+   * package building according to PR branch.
+   * branched and removed automatically from `systemsmanagement:saltstack:github/salt` by OBS workflow.
+
+The same OBS structure will apply to all our OBS targets: `products`, `products:testing` and `products:next`.
+
+### Packaging artifacts
+
+All current extra "Sources" files in our RPM package, together with spec file and changelog file will go now to a `pkg/suse/` directory in `openSUSE/salt`:
+
+```
+pkg/suse/README.SUSE
+pkg/suse/html.tar.bz2
+pkg/suse/salt-tmpfiles.d
+pkg/suse/transactional_update.conf
+pkg/suse/update-documentation.sh
+pkg/suse/salt.spec
+pkg/suse/salt.changes
+```
+
+This is the place now where all those files will be maintained.
+
+#### Salt RPM changelog
+
+As mentioned this is now at `pkg/suse/salt.changes` in `openSUSE/salt` GitHub repo.
+
+When creating a PR to `openSUSE/salt` the user must also include the corresponding changes to the changes file, that can be generated as usual with `osc vc`.
+
+Similarly to the main Uyuni repository, we should add a GitHub action to warn the user in case no changelog entry is added in the PR.
+
+### OBS project structure
+
+#### `systemsmanagement:saltstack:github/salt`
+
+This OBS package will only contain `_multibuild` file and a `_service` file that should look like: 
+
+```
+<services>
+  <service name="obs_scm">
+    <param name="url">https://github.com/openSUSE/salt.git</param>
+    <param name="scm">git</param>
+    <param name="versionformat">@PARENT_TAG@</param>
+    <param name="versionrewrite-pattern">v(.*)</param>
+    <param name="revision">openSUSE/release/xxxx</param>
+    <param name="extract">pkg/suse/salt.*</param>
+  </service>
+  <service name="set_version" />
+  <service name="tar" mode="buildtime"/>
+  <service name="recompress" mode="buildtime">
+    <param name="file">*.tar</param>
+    <param name="compression">gz</param>
+  </service>
+</services>
+```
+
+The rest of the files will be automatically pulled by the service, as they are enabled here. This package will be automatically refreshed by OBS at any new commit at `openSUSE/release/xxxx` branch.
+
+#### `systemsmanagement:saltstack/salt`
+
+This OBS package is a branch from `systemsmanagement:saltstack:github/salt`, where we disable the services and manually run them to get the spec file, changelog and obsinfo/obscpio files, so the package can be submitted to openSUSE or SLE.
+
+The `_service` file should look like:
+
+```
+<services>
+  <service name="obs_scm" mode="manual">
+    <param name="url">https://github.com/openSUSE/salt.git</param>
+    <param name="scm">git</param>
+    <param name="versionformat">@PARENT_TAG@</param>
+    <param name="versionrewrite-pattern">v(.*)</param>
+    <param name="revision">openSUSE/devel/master</param>
```

Comment:
I was not aware we used an `openSUSE/devel/master` branch. Is this a new branch we will use?

Reply:
I used `openSUSE/devel/master` here just as an example, it should point to the eventual `openSUSE/release/3008.x` branch.

Currently `openSUSE/devel/master` is just the devel branch I created with upstream `master` branch + our patches partially rebased on top (excluding patches to extensions).